lib/src/parser/reader.dart

```diff
@@ -50,8 +75,113 @@ class NgTokenReader {
   }
 
   /// Returns the next token without incrementing.
-  NgToken peek() => _peek = next();
+  /// Returns null otherwise.
+  NgBaseToken peek() => _peek = next();
+
+  /// Returns the next token type without incrementing.
+  /// Returns null otherwise.
+  NgBaseTokenType peekType() {
+    _peek = next();
+    if (_peek != null) {
+      return _peek.type;
+    }
+    return null;
```

Comment:
This reader is meant to be used generically for both NgToken and NgSimpleToken - so null would be the best option still. However, I definitely want to revisit this two layer Token system and integrate them into a more uniform way - in which case I can return a single EOF token.